site/src/layouts/ProjectSelector.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Link from 'gatsby-link'
+import styled from 'emotion/react'
+import cx from 'classnames'
+import { monofont, sansfont, baseUl } from './emotion-base'
+
+const Container = styled.div`
+  padding: 25px 20px;
+  border: 2px solid #000;
+`
+
+const ProjectSection = styled.section`
+  &:not(:last-child) {
+    margin-bottom: 30px;
+  }
+`
+
+const Year = styled.h3`
+  composes: ${monofont};
+  font-weight: bold;
+  font-size: 22px;
+  padding: 0;
+  margin: 0;
+`
+
+const ProjectList = styled.ul`
+  composes: ${baseUl}, ${sansfont};
+  display: flex;
+  flex-wrap: wrap;
+  margin-top: 16px;
+  font-size: 16px;
+
+  & li {
+    padding: 2px;
+    width: 50%;
+
+    &.active a {
+      background: #ff0;
+      text-decoration: underline;
+    }
+
+    & a {
+      color: inherit;
+      text-decoration: inherit;
+
+      &:hover,
+      &:focus {
+        text-decoration: underline;
+      }
+    }
+  }
+`
+
+const ProjectSelector = ({ projects, currentProjectSlug }) => {
+  const projectsByYear = Array.from(new Set(projects.map(p => p.when))) // years
+    .sort((a, b) => b.localeCompare(a)) // sort reverse-chronologically
```

Comment:
Nit: Why can't a numeric comparison be used here?

Reply:
good point, fixed :)